CMakeLists.txt

```diff
@@ -25,25 +30,29 @@ if(NOT MSVC)
   unset(compiler_id_lower)
 
   if(FORCE_COLORED_OUTPUT)
-    if(CMAKE_COMPILER_IS_GNUCC)
-      add_compile_options (-fdiagnostics-color=always)
-    elseif(D3_COMPILER_IS_CLANG)
-      add_compile_options (-fcolor-diagnostics)
+    if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.24)
+      set(CMAKE_COLOR_DIAGNOSTICS ON)
+    else ()
+      if(CMAKE_COMPILER_IS_GNUCC)
+        add_compile_options (-fdiagnostics-color=always)
+      elseif(D3_COMPILER_IS_CLANG)
+        add_compile_options (-fcolor-diagnostics)
+      endif ()
```

Comment:
We can just update `cmake_minumum_required` to 3.24 and simplify block to just `set(CMAKE_COLOR_DIAGNOSTICS ON)`.
I'd prefer remove this block entirely (and compiler detection above) as color output can be enabled just by `-DCMAKE_COLOR_DIAGNOSTICS=ON` on config time.

Reply:
thanks, it's better now. However the Windows CI build seems to fail now